### PR TITLE
Modified rcu script to support both SOA and FMWInfra Domain

### PR DESCRIPTION
**Pull request by anpanigr**
Updated the sample rcu creation scripts to work for both FMW and SOA Infra domain.  
Following modification were made to the scripts create-rcu-schema.sh/drop-rcu-schema.sh 
in the directory kubernetes/samples/scripts/create-rcu-schema
 
No change to the script interface. Based on the image String, the script will add the extra components and variables while creating the rcu for  FMW and SOA Infra  using the following logic

======= create-rcu-schema.sh ====
[[ $fmwimage =~ .*soasuite.* ]] && rcuType=soa
[[ $fmwimage =~ .*fmw-infr.* ]] && rcuType=fmw
=========
Starting of oracle DB Service remains same using the script start(stop)-db-service.sh

I run QUICK test on my branch to see any Regression on sample
http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/2784/console
All Tests Clean

### CODE REVIEW

File: kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh [hunk 1]
```diff
@@ -7,7 +7,9 @@
 echo "Check if the DB Service is ready to accept request "
 connectString=${1:-oracle-db.default.svc.cluster.local:1521/devpdb.k8s}
 schemaPrefix=${2:-domain1}
-echo "DB Connection URL [$connectString] and schemaPrefix is [${schemaPrefix}]"
+rcuType=${3:-fmw}
```
Comment (markxnelson) [inline kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh]: you are introducing a new parameter right? seems like there should be a matching documentation update in this PR too please

File: kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh [hunk 2]
```diff
@@ -27,10 +29,25 @@ else
  java utils.dbping ORACLE_THIN scott tiger ${connectString}
 fi
 
+# SOA need extra component(s) SOAINFRA and ESS
```
Comment (markxnelson) [inline kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh]: need -> needs

File: kubernetes/samples/scripts/create-rcu-schema/common/dropRepository.sh [hunk 3]
```diff
@@ -27,10 +29,21 @@ else
  java utils.dbping ORACLE_THIN scott tiger ${connectString} 
 fi 
 
+# SOA need extra component(s) SOAINFRA and ESS
```
Comment (markxnelson) [inline kubernetes/samples/scripts/create-rcu-schema/common/dropRepository.sh]: need -> needs

File: kubernetes/samples/scripts/create-rcu-schema/create-rcu-schema.sh [hunk 4]
```diff
@@ -21,7 +21,7 @@ function usage {
   exit $1
 }
 
-while getopts ":h:s:d:p:" opt; do
+while getopts ":h:s:d:p:i:" opt; do
```
Comment (markxnelson) [inline kubernetes/samples/scripts/create-rcu-schema/create-rcu-schema.sh]: should be doc for this new `i` option you are adding

Comment (anpanigr) [reply kubernetes/samples/scripts/create-rcu-schema/create-rcu-schema.sh]: Actually it was a typo in the existing script for image input parameter.  The description of the parameter is already there.

<bash>./create-rcu-schema.sh 
create-rcu-schema.sh: -s <schemaPrefix> must be specified.
usage: create-rcu-schema.sh -s <schemaPrefix> -d <dburl> -i <image> -p <docker-store> [-h]
  -s RCU Schema Prefix (needed)
  -d RCU Oracle Database URL (optional) 
      (default: oracle-db.default.svc.cluster.local:1521/devpdb.k8s) 
  -p FMW Infrastructure ImagePull Secret (optional) 
      (default: docker-store) 
  -i FMW Infrastructure Image (optional) 
      (default: container-registry.oracle.com/middleware/fmw-infrastructure:12.2.1.3) 
  -h Help

File: kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh (hunk 1 above)
Comment (anpanigr) [reply kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh]: This input parameter (rcuType) is determined automatically in the parent script  create-rcu-schema.sh as described in the description of PR. Here we are not adding new input parameter but determine the rcuType form the rcu image.

File: kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh (hunk 2 above)
Comment (anpanigr) [reply kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh]: resolved

File: kubernetes/samples/scripts/create-rcu-schema/common/dropRepository.sh (hunk 3 above)
Comment (anpanigr) [reply kubernetes/samples/scripts/create-rcu-schema/common/dropRepository.sh]: resolved

File: kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh (hunk 1 above)
Comment (markxnelson) [reply kubernetes/samples/scripts/create-rcu-schema/common/createRepository.sh]: ok

**Comment by anpanigr** [comment]
HI Vivek,

I have added a PR https://github.com/oracle/weblogic-kubernetes-operator/pull/1298 
for the suggested change

**Pull request merged**